javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/typesolvers/ClassLoaderTypeSolver.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2016-2018 The JavaParser Team.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.javaparser.symbolsolver.resolution.typesolvers;
+
+import com.github.javaparser.resolution.declarations.ResolvedReferenceTypeDeclaration;
+import com.github.javaparser.symbolsolver.model.resolution.SymbolReference;
+import com.github.javaparser.symbolsolver.model.resolution.TypeSolver;
+import com.github.javaparser.symbolsolver.reflectionmodel.ReflectionFactory;
+
+import java.util.Optional;
+
+/**
+ * This TypeSolver wraps a ClassLoader. It can solve all types that the given ClassLoader can load.
+ * This is intended to be used with custom classloaders. To support typical cases based on reflection
+ * just use the ReflectionTypeSolver
+ *
+ * @author Federico Tomassetti
+ */
+public class ClassLoaderTypeSolver implements TypeSolver {
+
+    private TypeSolver parent;
+    private ClassLoader classLoader;
+
+    public ClassLoaderTypeSolver(ClassLoader classLoader) {
+        this.classLoader = classLoader;
+    }
+
+    @Override
+    public TypeSolver getParent() {
+        return parent;
+    }
+
+    @Override
+    public void setParent(TypeSolver parent) {
+        this.parent = parent;
+    }
+
+    protected boolean filterName(String name) {
+        return true;
+    }
+
+    @Override
+    public SymbolReference<ResolvedReferenceTypeDeclaration> tryToSolveType(String name) {
```

Comment:
If JavaParser API is invoked to resolve the class (for e.g my.company.SampleClass) then this method also receive the input call where **name** param contains value "java.lang.my.company.SampleClass".

Reply:
Perhaps performance improvement, if this type of redundant call can be fixed.

Reply:
I am not sure I understand

Reply:
I will create separate ticket with more info as this issue is applicable to ReflectionTypeResolver too, btw this PR looks good to me for Classloader type resolver support.
-Thanks

Reply:
Ok, then I will merge this one and work on the issues you will submit later. Thanks!